### ISSUE TITLE
New Check: DuplicateSwitchBranchCheck

### ISSUE THREAD

**Issue by MonuChaudhary14** [checkstyle/checkstyle#20571]
## Description

Checkstyle currently does not provide a check that detects duplicate branches in `switch` statements or `switch` expressions.

## Proposed Check

`DuplicateSwitchBranchCheck`

The check should report `switch` branches whose implementations are identical.

## Violation Example

```java
class Test {

    int getNumber(String value) {
        switch (value) {
            case "A":
                return 1;

            case "B":
                return 1;

            case "C":
                return 2;

            default:
                return 0;
        }
    }
}
```

Expected violation:

```
Switch branches for cases "A" and "B" have identical implementations.
```

Switch expression example:

```java
class Test {

    int getNumber(String value) {
        return switch (value) {
            case "A" -> 1;
            case "B" -> 1;
            case "C" -> 2;
            default -> 0;
        };
    }
}
```

Expected violation:

```
Switch branches for cases "A" and "B" have identical implementations.
```

## Non-Violation Example

```java
class Test {

    int getNumber(String value) {
        switch (value) {
            case "A":
            case "B":
                return 1;

            case "C":
                return 2;

            default:
                return 0;
        }
    }
}
```

Switch expression equivalent:

```java
class Test {

    int getNumber(String value) {
        return switch (value) {
            case "A", "B" -> 1;
            case "C" -> 2;
            default -> 0;
        };
    }
}
```

## Expected Configuration

```xml
<module name="Checker">
    <module name="TreeWalker">
        <module name="DuplicateSwitchBranch"/>
    </module>
</module>
```

## What should NOT be flagged

- Multiple case labels sharing the same branch.

```java
case 1:
case 2:
    doWork();
    break;
```

```java
case 1, 2 -> doWork();
```

- Branches with different implementations.

```java
case 1:
    log();
    return 1;

case 2:
    return 1;
```

- Intentional fall-through.

```java
case 1:
    prepare();

case 2:
    execute();
    break;
```

**Comment by MonuChaudhary14** [comment]
@romani 
Please look to this

**Comment by romani** [comment]
approved
